puppet/hiera/hiera.yaml

```diff
@@ -1,13 +1,21 @@
 ---
 :backends:
-        - yaml
+  - yaml
 :yaml:
-        :datadir: '/usr/local/puppet/hiera/'
+  :datadir: '/usr/local/puppet/hiera/'
 :hierarchy:
-        - "nodes/external/secret/%{::fqdn}"
-        - "nodes/external/%{::fqdn}"
-        - "nodes/vagrant-%{::vagrant_extra_prefix}%{::is_vagrant}"
-        - "nodes/common"
-        - "vagrant-%{::is_vagrant}"
-        - "common"
-:merge_behavior: deeper
+  # things in nodes/ are part of event-specific repositories, not included in this repo
+  - "nodes/external/secret/%{::fqdn}"
+  - "nodes/external/secret/event-%{::event_name}"
+  - "nodes/external/secret/%{::environment}"
+  - "nodes/external/%{::fqdn}"
+  - "nodes/vagrant-%{::event_name}-%{::is_vagrant}"
+  - "nodes/vagrant-%{::is_vagrant}"
+  - "nodes/event-%{::event_name}"
+  - "nodes/%{::environment}"
+  - "nodes/common"
+  # things below nodes are included in this repo and visible to any event running uber
+  - "vagrant-%{::is_vagrant}"
+  - "%{::environment}"
+  - "common"
+# :merge_behavior: deeper # used to need this, probably not anymore.
```

Comment:
So the way this works is, each server is, before puppet is run, assigned a set of 'facts' through a program called Facter.

In the case of let's say, stage.uber.magfest.org, the facts would be:
event_name='classic'
environment='development'
is_vagrant='false'
fqdn='stage.uber.magfest.org' (fully qualified domain name)

For prime.uber.magfest.org it would be
event_name='prime'
environment='production'
is_vagrant='false'
fqdn='prime.uber.magfest.org'

For a local vagrant instance, it would be:
event_name='[whatever you want it to be]'
environment='development'
is_vagrant='true'
fqdn='localhost.localdomain'

These facts are used to include certain files according to hiera.yaml.  Any settings set in both a lower file and a higher file is hidden by stuff in the higher up file:
```
"nodes/external/secret/%{::fqdn}"
"nodes/external/secret/event-%{::event_name}-%{::environment}"
"nodes/external/secret/event-%{::event_name}"
"nodes/external/secret/%{::environment}"
"nodes/external/secret/common"
"nodes/external/%{::fqdn}"
"nodes/vagrant-%{::event_name}-%{::is_vagrant}"
"nodes/vagrant-%{::is_vagrant}"
"nodes/event-%{::event_name}"
"nodes/%{::environment}"
"nodes/common"
"vagrant-%{::is_vagrant}"
"%{::environment}"
"common"
```

(note: for this example I added a few entries not present cause I haven't gotten to them yet)

For stage.uber.magfest.org it looks like this:
```
"nodes/external/secret/stage.uber.magfest.org.yaml"
"nodes/external/secret/event-classic-development.yaml"
"nodes/external/secret/event-classic.yaml"
"nodes/external/secret/development.yaml"
"nodes/external/secret/common.yaml"
"nodes/external/stage.uber.magfest.org.yaml"
"nodes/vagrant-classic-0.yaml"
"nodes/vagrant-0.yaml"
"nodes/event-classic.yaml"
"nodes/development.yaml"
"nodes/common.yaml"
"vagrant-0.yaml"
"development.yaml"
"common.yaml"
```

Not all of those files exist (for instance, vagrant-0.yaml never exists and stage.uber,magfest.org,yaml) so here's a list of just the ones that do exist or aren't placeholders:
```
"nodes/external/secret/event-classic-development.yaml"
"nodes/external/secret/common.yaml"
"nodes/external/secret/development.yaml"
"nodes/event-classic.yaml"
"nodes/common.yaml"
"common.yaml"
```

Going node by node here for that stuff:
common.yaml - default settings we expect all events to use.  things like system settings, database access control settings, etc.  which core ubersystem plugins to use (uber, analytics, etc)
nodes/ -> everything in here is magfest-specific and only included by magfest.
nodes/common.yaml -> magfest-specific default settings.  things like email signatures, magfest shirt sizes, SSL certificates for magfest.org, organization_name='Magfest INC'.  basically anything in common between magprime, magclassic, magstock, rockage
nodes/event-classic.yaml -> the vast majority of magfest-specific settings. badge prices, dates, food prices, dept head checklists, any event-specific plugins to install etc
nodes/external/secret/common.yaml -> secret settings common to all magfest events.  for us, it's AWS keys.
nodes/external/secret/event-classic-development.yaml -> Stripe keys - testing versions only



For other events like Anthrocon, we'd expect them to supply a completely new nodes/ dir and plug in their specific settings there.

Anyway, I think everyone's going to find that these hierarchy settings make a ton of sense and make adding changes easier.  What's really cool is that, for example, vagrant now uses the exact same settings as a staging server, without any extra work copying a bunch of junk around.

It's a little tough to read these changes in the diff, I do suggest grabbing a copy of these files in magfest/production-config and just browsing them there.